Frameworks/Foundation/NSProgress.mm

```diff
@@ -208,7 +209,9 @@ - (void)resignCurrent {
         // If you don’t create any child progress objects between the calls to becomeCurrentWithPendingUnitCount: and resignCurrent,
         // the "parent" progress automatically updates its completedUnitCount by adding the pending units.
         if (!currentProgressStack->top().childCreated) {
```

Comment:
>currentProgressStack->top() [](start = 13, length = 27)

Doesn't this expression evaluate to NSProgress currentProgress? It might make sense to cache it and use it in this function?